evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -74,6 +74,9 @@ pub struct Cheatcodes {
     /// Recorded storage reads and writes
     pub accesses: Option<RecordAccess>,
 
+    /// Recorded logs
+    pub recorded_logs: Option<RecordedLogs>,
```

Comment:
Might be better to have a boolean indicating whether we are recording or not so you can do something like:

```solidity
vm.recordLogs();
// some calls (lets say call a, b and c)
vm.getRecordedLogs(); // gets logs for calls a, b and c
// perform call d and call e
vm.getRecordedLogs(); // gets logs for calls d and e
```

Additionally I think a call to `vm.getRecordedLogs` should only return logs between the call to `getRecordedLogs` and the previous call to `recordLogs` or `getRecordedLogs` to allow for the pattern above. Currently this is the behavior:

```solidity
vm.recordLogs();
// some calls (lets say call a, b and c)
vm.getRecordedLogs(); // gets logs for calls a, b and c
// perform call d and call e
vm.getRecordedLogs(); // gets logs for  calls a, b, c, d and e
```

To discard logs for the first 3 calls you would have to call `recordLogs` again to reset the buffer